vault/resource_kv_secret_test.go

```diff
@@ -69,3 +83,46 @@ resource "vault_kv_secret" "test" {
 
 	return ret
 }
+
+func testKVSecretConfig_updated(mount, name string) string {
+	ret := fmt.Sprintf(`
+%s
+
+`, kvV1MountConfig(mount))
+
+	ret += fmt.Sprintf(`
+resource "vault_kv_secret" "test" {
+  path = "${vault_mount.kvv1.path}/%s"
+  data_json = jsonencode(
+    {
+      zip = "zoop",
+      foo = "bar"
+    }
+  )
+}`, name)
+
+	return ret
+}
+
+func testResourceKVSecret_apiAcessCheck(s *terraform.State) error {
```

Comment:
Please adapt this check to follow the pattern set in: https://github.com/hashicorp/terraform-provider-vault/blob/c520061319a277600ea367adacdcf7e17d2820ee/vault/resource_identity_entity_test.go#L143

Reply:
nice pattern!

Reply:
done with
[refactor api access check to functional pattern](https://github.com/hashicorp/terraform-provider-vault/pull/1570/commits/e23506b34a074546367933bec992886108f4e196)